specs/ContextMenuRequested.md

```diff
@@ -0,0 +1,477 @@
+
+ ```c#
+    webView.CoreWebView2.ContextMenuRequested += delegate (object sender, CoreWebView2ContextMenuRequestedEventArgs args)
+    {
+        CoreWebView2Deferral deferral = args.GetDeferral();
+        System.Threading.SynchronizationContext.Current.Post((_) =>
+        {
+            using (deferral)
+            {
+                IList<CoreWebView2ContextMenuItem> menuList = args.MenuItems;
+                CoreWebView2SelectionContext context = args.context;
+                args.Handled = true;
+                ContextMenu cm = this.FindResource("ContextMenu") as ContextMenu;
+                cm.Items.Clear();
+                for(int i = 0; i < menuList.Count; i ++){
+                    CoreWebView2ContextMenuItem current = menuList[i];
+                    MenuItem newItem = new MenuItem();
+                    newItem.Header = current.Name;
+                    newItem.Click += (s, ex) => args.SelectedItem = current.id;
+                    cm.Items.Add(newItem);
+                }
+                cm.IsOpen = true;
+            }
+        }, null);
+    };
+```
+# Remarks
+
+# API Notes
+
+# API Details
+ ```cpp
+    interface ICoreWebView2_4;
+    interface ICoreWebView2Environment;
+    interface ICoreWebView2ContextMenuItem;
+    interface ICoreWebView2ContextMenuItemCollection;
+    interface ICoreWebView2ContextMenuRequestedEventArgs;
+    interface ICoreWebView2ContextMenuRequestedEventHandler;
+
+    /// Defines the Context Menu Items
+    [v1_enum]
+    typedef enum COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR {
+        /// Custome type, will include Items brought in by browser extensions
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_CUSTOM,
+
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_MISSPELLED,
+
+        /// No Command
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_NONE,
+
+        /// Page Commands
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_BACK,
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_FORWARD,
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_RELOAD,
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_STOP,
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_NEW_WINDOW,
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_PRINT,
+
+        /// Clipboard Commands
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_CUT,
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_COPY,
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_PASTE,
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_PASTE_AS_PLAIN_TEXT,
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_SELECT_ALL,
+
+        /// Saving
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_SAVE_LINK_AS,
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_SAVE_IMAGE_AS,
+
+        COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR_INSPECT, 
+        
+    } COREWEBVIEW2_CONTEXT_MENU_ITEM_DESCRIPTOR;
+    
+    /// Indicates the context selected (where the user right clicked)
+    [v1_enum]
+    typedef enum COREWEBVIEW2_SELECTION_CONTEXT {
```

Comment:
It seems like this is missing other contexts like links, or text. Why are some things in this list and not others?